File: examples/sql_injection/after.js
// After: Parameterized query
const db = require('./db');
app.get('/users', (req, res) => {
  const query = 'SELECT * FROM users WHERE name = ?';
  db.query(query, [req.query.name], (err, results) => {
    if (err) console.error(err);
    res.send(results);
  });
});

File: examples/sql_injection/after.py
import sqlite3

def get_user(user_id):
    conn = sqlite3.connect('example.db')
    cursor = conn.cursor()
    # FIXED: parameterized query prevents SQL injection
    query = "SELECT * FROM users WHERE id = ?"
    cursor.execute(query, (user_id,))
    return cursor.fetchall()